crates/toml_edit/src/encode.rs

```diff
@@ -455,16 +455,11 @@ impl StringStyle {
 }
 
 fn infer_style(value: &str) -> (StringStyle, bool) {
-    // For doing pretty prints we store in a new String
-    // because there are too many cases where pretty cannot
-    // work. We need to determine:
+    // We need to determine:
     // - if we are a "multi-line" pretty (if there are \n)
     // - if ['''] appears if multi or ['] if single
     // - if there are any invalid control characters
-    //
-    // Doing it any other way would require multiple passes
-    // to determine if a pretty string works or not.
```

Comment:
Isn't this part of the comment still relevant?

Reply:
I'm not sure. The comment seems to justify why a string was being built. It also explains what the function does, which is still true. I tried to get rid of the justification for the string and keep the rest.

Happy to keep those two lines if you think that's better.

Reply:
I think its referring to why the heuristics are so limited but not the biggest deal.

Reply:
Thanks for explaining. I've brought the two lines back.